mmv1/products/accesscontextmanager/api.yaml

```diff
@@ -899,6 +905,220 @@ objects:
                 The list of APIs usable within the Service Perimeter.
                 Must be empty unless `enableRestriction` is True.
               item_type: Api::Type::String
+          - !ruby/object:Api::Type::Array
+            name: 'ingressPolicies'
+            description: /
+              List of `IngressPolicies` to apply to the perimeter. A perimeter may
+              have multiple `IngressPolicies`, each of which is evaluated
+              separately. Access is granted if any `Ingress Policy` grants it.
+              Must be empty for a perimeter bridge.
+            at_least_one_of:
+              - status.0.resources
+              - status.0.access_levels
+              - status.0.restricted_services
+              - status.0.ingress_policies
+              - status.0.egress_policies
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
+              - !ruby/object:Api::Type::NestedObject                  
+                name: 'ingressFrom'
+                description: /
+                  Defines the conditions on the source of a request causing this `IngressPolicy`
+                  to apply.
+                at_least_one_of:
+                  - status.0.ingress_policies.ingressFrom
+                  - status.0.ingress_policies.ingressTo
+                properties:
+                - !ruby/object:Api::Type::Enum
+                  name: 'identityType'
+                  description: /
+                    Specifies the type of identities that are allowed access from outside the 
+                    perimeter. If left unspecified, then members of `identities` field will be 
+                    allowed access.
+                  values:
+                    - :IDENTITY_TYPE_UNSPECIFIED
+                    - :ANY_IDENTITY
+                    - :ANY_USER_ACCOUNT
+                    - :ANY_SERVICE_ACCOUNT
+                - !ruby/object:Api::Type::Array
+                  name: 'identities'
+                  item_type: Api::Type::String
+                  description: /
+                    A list of identities that are allowed access through this ingress policy.
```

Comment:
Is there a reason this would ever be empty or not specified? Can we make this required: true?

Reply:
The API allows you to specify either `identities` or `identity_type`, so I think what's needed is an `exactly_one_of`. The same goes for `egress_policies.egressFrom`.

Reply:
This was how I had the validation properties set on a PR that was trying to do the same thing as this one: https://github.com/GoogleCloudPlatform/magic-modules/pull/4535/commits/23baeaec9428e9cc11428c6e9c66d040d866790a